### PR TITLE
docs: rewrite Observable examples with pipable operators

### PR DESCRIPTION
**Pull request by lacolaco**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/31024

Observable examples are outdated.

## What is the new behavior?

Using up-to-date pipeable operators

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/comparing-observables.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ An observable produces values over time. An array is created as a static set of
   <tr>
     <td><pre>concat()</pre></td>
     <td>
-      <pre>obs.concat(obsB)</pre>
+      <pre>concat(obs, obsB)</pre>
```
Comment (lacolaco) [inline aio/content/guide/comparing-observables.md]: `concat` operator is deprecated. Currently, static `concat` creator function is only recommended.

File: aio/content/guide/comparing-observables.md [hunk 2]
```diff
@@ -305,7 +305,7 @@ An observable produces values over time. An array is created as a static set of
   <tr>
     <td><pre>reduce()</pre></td>
     <td>
-      <pre>obs.scan((s,v)=> s+v, 0)</pre>
+      <pre>obs.pipe(scan((s,v)=> s+v, 0))</pre>
```
Comment (brandonroberts) [inline aio/content/guide/comparing-observables.md]: Separate issue, but this seems odd that we aren't using `reduce` here.

**Review by brandonroberts — APPROVED** [review]

File: aio/content/guide/comparing-observables.md (hunk 2 above)
Comment (lacolaco) [reply aio/content/guide/comparing-observables.md]: indeed. `➞1➞3➞6➞11➞18` is not corresponding to array-side result (`18`).

Comment (brandonroberts) [reply aio/content/guide/comparing-observables.md]: Will you go ahead and change it to use `reduce` instead?

**Comment by lacolaco** [comment]
Sure. give me a day.

2019年6月18日(火) 3:27 Brandon <notifications@github.com>:

File: aio/content/guide/comparing-observables.md (hunk 2 above)
Comment (lacolaco) [reply aio/content/guide/comparing-observables.md]: Sure. give me a day.

Comment (lacolaco) [reply aio/content/guide/comparing-observables.md]: @brandonroberts I think it is not bad to include the change in this PR. should I separate it?

Comment (brandonroberts) [reply aio/content/guide/comparing-observables.md]: You can include it in this one

**Comment by brandonroberts** [comment]
@lacolaco you will to rebase on latest master to fix the CI failure, and squash down to a single commit.

**Comment by brandonroberts** [comment]
Caretaker note: I am using a global approval for docs. The Codefresh failure is unrelated

**Pull request closed without merging**